src/components/kubectl/logs.ts

```diff
@@ -127,7 +128,8 @@ async function getLogsForContainer(
         if (!result || result.code !== 0) {
             vscode.window.showErrorMessage(`Error reading logs: ${result ? result.stderr : undefined}`);
         } else {
-            panel.setInfo(result.stdout, containerResource.kindName);
+            panel.deleteAppendContentProcess();
```

Comment:
Not sure what is going on here.  In this code path the `kubectl` call exits, so there is no child process to delete.  Does this relate to the case where the user was previously doing Follow Logs for a pod and has now done a Show Logs, and we are trying to reuse the webview so we have to clear out the previous follow-related state?  If so perhaps explain this in a comment?

I kind of feel that maybe this shouldn't be a client problem.  If the webview is hanging onto state, then the webview should be responsible for cleaning it up - perhaps the LogsPanel should have two distinct methods, one to start a follow and one to perform a show, and those methods should handle the necessary state management?

Reply:
The use case you described it's exactly what's happening here. Going from a `Follow Logs` to a `Show Logs` and reuse the webview.

I don't follow what you're saying about the two distinct method to handle the state management. There is no much to handle here. If it's a `Follow Logs` we just need to keep a reference of the child process created (so it can be killed later), if it's a `Show Logs` we need to kill the old child process if it exists and this is made by calling `setAppendContentProcess` and `deleteAppendContentProcess`.